admin_pages/messages/Messages_Admin_Page.core.php

```diff
@@ -1308,6 +1308,10 @@ protected function _edit_message_template()
             ? absint($this->_req_data['id'])
             : false;
 
+        $EVT_ID = isset($this->_req_data['evt_id']) && ! empty($this->_req_data['evt_id'])
+        ? absint($this->_req_data['evt_id'])
+        : false;
```

Comment:
although PHP allows it, we want to avoid mixing data types in vars as much as possible because it helps to avoid an entire category of bugs that result from mismatched types.
So instead of mixing integers and booleans here, it's better to just stick with integers and use `0` instead of `false` which should still work the same in most conditionals since it evaluates as "falsey"

ie:

```php
        $EVT_ID = isset($this->_req_data['evt_id']) && ! empty($this->_req_data['evt_id'])
        ? absint($this->_req_data['evt_id'])
        : 0;
```

that said, you should verify (if you haven't already) that the models or any other logic downstream that will be consuming the results of this code, are NOT performing a strict comparison and expecting to see `false`

ie:

```php
if ( isset($this->_req_data['evt_id'] && $this->_req_data['evt_id'] === false ) { // do something }
```

which is easy enough to fix simply by replacing the strict comparison with a loose comparison:

```php
if ( isset($this->_req_data['evt_id'] && $this->_req_data['evt_id'] ) { // do something }
```

or keeping things strict but incorporating type conversion

```php
if ( isset($this->_req_data['evt_id'] && (int)$this->_req_data['evt_id'] > 0 ) { // do something }
```

or 

```php
if ( isset($this->_req_data['evt_id'] && absint($this->_req_data['evt_id']) > 0 ) { // do something }
```

or ... 

```php
// insert your preferred approach here
```

Reply:
Whilst I don't have any issues with changing the above to always be an int, I went with using `false` because that's how it's set up for `GRP_ID` just above. I did dig into it a little previously and I think the reason it currently uses `false` is because we use [add_query_arg()](https://developer.wordpress.org/reference/functions/add_query_arg/) to build out the query string.

Setting a query var's value to false, removes the pair from the string and so avoids things like:

`admin.php?page=espresso_messages&message_type=registration&messenger=email&context=attendee&GRP_ID=3&evt_id=0&action=preview_message&preview_message_nonce=0608dc82cf`

For the default preview links when editing them through Event Espresso -> Messages.

The rest of the system handles `evt_id=0` just fine but I could add some conditionals to remove it if preferred?

Reply:
ahh ok... it didn't occur to me to consider `add_query_arg()` 

that's perfectly fine then as is, thnx for explaining.

Reply:
Thanks to Brents eagle eye I found a notice thrown when previewing messages from the default message templates editor, no evt_id is set on that request but I was trying to use it.